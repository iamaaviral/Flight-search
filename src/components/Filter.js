import React, { Component } from 'react';

class Filter extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      minValue: 0,
      maxValue: 5000
    };
    this.minHandleChange = this.minHandleChange.bind(this);
    this.maxHandleChange = this.maxHandleChange.bind(this);
    this.search = this.search.bind(this);
  }
  minHandleChange(event) {
    this.setState({ minValue: event.target.value });
  }
  maxHandleChange(event) {
    this.setState({ maxValue: event.target.value });
  }

  search(e) {
    e.preventDefault();
    console.log('min: ' + this.state.minValue + 'max: ' + this.state.maxValue);
  }

  render() {
    return (
      <div>
        <div>
          <span>0</span>
          <input
            id="typeinp"
            type="range"
            min="0"
            max="50000"
            value={this.state.minValue}
            onChange={this.minHandleChange}
            step="500"
          />
          <span>50000</span>
        </div>
        <div>
          <span>{this.state.minValue}</span>
          <input
            id="typeinp"
            type="range"
            min={this.state.minValue}
            max="50000"
            value={this.state.maxValue}
            onChange={this.maxHandleChange}
            step="500"
          />
          <span>50000</span>
        </div>
        <div>
          {`min: ₹${this.state.minValue} `}
          {` max: ₹${this.state.maxValue}`}
        </div>
        <div className="btn-box">
          <button
            className="button btn-submit"
            onClick={() =>
              this.props.onFilter(this.state.minValue, this.state.maxValue)
            }
            type="filter"
          >
            FILTER
          </button>
        </div>
      </div>
    );
  }
}

export default Filter;
