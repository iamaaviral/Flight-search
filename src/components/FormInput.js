import React, { Component } from 'react';
import DatePicker from 'react-datepicker';
import moment from 'moment';
import myData from '../flights.json';

import 'react-datepicker/dist/react-datepicker.css';

class FlightForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      from: '',
      to: '',
      departureDate: moment()
      //returnDate: moment().add(2, "days")
    };
    this.search = this.search.bind(this);
    this.change = this.change.bind(this);
    this.handleChange = this.handleChange.bind(this);
  }

  handleChange(date) {
    this.setState({
      departureDate: date
    });
  }

  change(e) {
    this.setState({ [e.target.name]: e.target.value });
    this.showInputError(e.target.name);
  }

  getFlightData() {
    let results = myData.filter((item, index) => {
      if (item.departure.iataCode !== this.state.from.toUpperCase()) {
        return;
      }
      if (item.arrival.iataCode !== this.state.to.toUpperCase()) {
        return;
      }
      if (
        moment.utc(item.departure.scheduledTime).format('MM/DD/YYYY') !==
        this.state.departureDate.format('MM/DD/YYYY')
      ) {
        return;
      }
      return item;
    });
    this.props.action(results);
  }

  search(e) {
    e.preventDefault();
    if (this.validate()) {
      this.getFlightData();
    }
  }

  validate() {
    const inputs = document.getElementById('input');
    let isFormValid = true;

    const isInputValid = this.showInputError(inputs.name);
    if (!isInputValid) {
      isFormValid = false;
    }
    return isFormValid;
  }

  showInputError(refName) {
    const validity = this.refs[refName].validity;
    const label = document.getElementById(`${refName}Label`).textContent;
    const error = document.getElementById(`${refName}Error`);

    if (!validity.valid) {
      if (validity.valueMissing) {
        error.textContent = `${label} is a required field`;
      }
      return false;
    }

    error.textContent = '';
    return true;
  }

  render() {
    return (
      <div className="wrapper">
        <form>
          <div className="group">
            <input
              id="input"
              type="text"
              name="from"
              ref="from"
              value={this.state.from}
              onChange={e => this.change(e)}
              required="required"
            />
            <span className="bar" />
            <label id="fromLabel">From</label>
            <div className="error" id="fromError" />
          </div>
          <div className="group">
            <input
              id="input"
              type="text"
              name="to"
              ref="to"
              value={this.state.to}
              onChange={e => this.change(e)}
              required="required"
            />
            <span className="bar" />
            <label id="toLabel">To</label>
            <div className="error" id="toError" />
          </div>
          <div className="group">
            <DatePicker
              name="date"
              selected={this.state.departureDate}
              onChange={this.handleChange}
              minDate={moment()}
              withPortal
              maxDate={moment().add(5, 'months')}
              showDisabledMonthNavigation
            />
          </div>
          {/* <div className="group">
            <DatePicker
            selected={this.state.returnDate}
            onChange={this.handleChange}
            minDate={moment()}
            withPortal
            maxDate={moment().add(5, "months")}
            showDisabledMonthNavigation
            />
          </div> */}
          <div className="btn-box">
            <button
              className="button btn-submit"
              onClick={e => this.search(e)}
              type="submit"
            >
              SEARCH
            </button>
          </div>
        </form>
      </div>
    );
  }
}

export default FlightForm;
