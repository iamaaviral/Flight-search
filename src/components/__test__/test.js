import React from 'react';
import Renderer from 'react-test-renderer';

import FormInput from '../FormInput';
import Filter from '../Filter';

describe('Tests for all screen', () => {
  it('should render FormInput without crashing', () => {
    const snap = Renderer.create(<FormInput />).toJSON();
    expect(snap).toMatchSnapshot();
  });
  
  it('should render Filter without crashing', () => {
    const snap = Renderer.create(<Filter />).toJSON();
    expect(snap).toMatchSnapshot();
  });

});