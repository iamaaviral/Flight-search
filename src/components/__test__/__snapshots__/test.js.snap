// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for all screen should render Filter without crashing 1`] = `
<div>
  <div>
    <span>
      0
    </span>
    <input
      id="typeinp"
      max="50000"
      min="0"
      onChange={[Function]}
      step="500"
      type="range"
      value={0}
    />
    <span>
      50000
    </span>
  </div>
  <div>
    <span>
      0
    </span>
    <input
      id="typeinp"
      max="50000"
      min={0}
      onChange={[Function]}
      step="500"
      type="range"
      value={5000}
    />
    <span>
      50000
    </span>
  </div>
  <div>
    min: ₹0 
     max: ₹5000
  </div>
  <div
    className="btn-box"
  >
    <button
      className="button btn-submit"
      onClick={[Function]}
      type="filter"
    >
      FILTER
    </button>
  </div>
</div>
`;

exports[`Tests for all screen should render FormInput without crashing 1`] = `
<div
  className="wrapper"
>
  <form>
    <div
      className="group"
    >
      <input
        id="input"
        name="from"
        onChange={[Function]}
        required="required"
        type="text"
        value=""
      />
      <span
        className="bar"
      />
      <label
        id="fromLabel"
      >
        From
      </label>
      <div
        className="error"
        id="fromError"
      />
    </div>
    <div
      className="group"
    >
      <input
        id="input"
        name="to"
        onChange={[Function]}
        required="required"
        type="text"
        value=""
      />
      <span
        className="bar"
      />
      <label
        id="toLabel"
      >
        To
      </label>
      <div
        className="error"
        id="toError"
      />
    </div>
    <div
      className="group"
    >
      <div>
        <div
          className="react-datepicker__input-container"
        >
          <input
            autoComplete={undefined}
            autoFocus={undefined}
            className=""
            disabled={false}
            id={undefined}
            name="departureDate"
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder={undefined}
            readOnly={false}
            required={undefined}
            tabIndex={undefined}
            title={undefined}
            type="text"
            value="09/30/2018"
          />
        </div>
      </div>
    </div>
    <div
      className="btn-box"
    >
      <button
        className="button btn-submit"
        onClick={[Function]}
        type="submit"
      >
        SEARCH
      </button>
    </div>
  </form>
</div>
`;
